src/System.Data.Common/src/System/Xml/XmlBoundElement.cs

```diff
@@ -397,7 +397,7 @@ public override XmlNodeList GetElementsByTagName(string name)
             // before iteration, so iteration will not cause foliation (and as a result of this, creation of new nodes).
             XmlNodeList tempNodeList = base.GetElementsByTagName(name);
 
-            int tempint = tempNodeList.Count;
+            _ = tempNodeList.Count;
```

Comment:
Why not remove this all together?

Reply:
Because the comment above, and complicated logic in some overrides, suggests that Count has a side effect that's important.  If someone in the know can confidently tell me otherwise, then I'll remove it.  But this code looks extremely deliberate, and the comment seems to hint at why.